backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/edi_desadv/ChangeEDI_ExportStatus_C_Doc_Outbound_Log_GridView.java

```diff
@@ -0,0 +1,140 @@
+	@Param(parameterName = PARAM_TargetExportStatus)
+	private String p_TargetExportStatus;
+
+	@Override
+	public ProcessPreconditionsResolution checkPreconditionsApplicable()
+	{
+		final ImmutableSet<ArchiveId> docOutboundLogIds = getSelectedDocOutboundLogIds();
+		if (docOutboundLogIds.isEmpty())
+		{
+			return ProcessPreconditionsResolution.rejectBecauseNoSelection();
+		}
+
+		// technical detail: all records must have the same Export Status
+		EDIExportStatus sameExportStatus = null;
+
+		for (final ArchiveId logId : docOutboundLogIds)
+		{
+			final I_C_Doc_Outbound_Log docOutboundLog = ediDocOutBoundLogService.retreiveById(logId);
+
+			if (!ChangeEDI_ExportStatusHelper.checkIsInvoiceAndEDI(docOutboundLog))
+			{
+				return ProcessPreconditionsResolution.rejectWithInternalReason("Selected record is not an EDI Invoice: " + docOutboundLog);
+			}
+			final EDIExportStatus fromExportStatus = EDIExportStatus.ofNullableCode(docOutboundLog.getEDI_ExportStatus());
+			if (fromExportStatus == null)
+			{
+				return ProcessPreconditionsResolution.rejectWithInternalReason("Selected record is not an EDI Invoice: " + docOutboundLog);
+			}
+
+			if (ChangeEDI_ExportStatusHelper.getAvailableTargetExportStatuses(fromExportStatus).isEmpty())
+			{
+				return ProcessPreconditionsResolution.rejectWithInternalReason("Cannot change ExportStatus from the current one: " + fromExportStatus);
+			}
+
+			if (sameExportStatus == null)
+			{
+				sameExportStatus = fromExportStatus;
+			}
+
+			if (!sameExportStatus.equals(fromExportStatus))
+			{
+				return ProcessPreconditionsResolution.rejectWithInternalReason("All records must have the same EDI ExportStatus");
+			}
+		}
+
+		return ProcessPreconditionsResolution.accept();
+	}
+
+	@ProcessParamLookupValuesProvider(parameterName = PARAM_TargetExportStatus, numericKey = false, lookupSource = LookupSource.list)
+	private LookupValuesList getTargetExportStatusLookupValues(final LookupDataSourceContext context)
+	{
+		final I_C_Doc_Outbound_Log docOutboundLog = ediDocOutBoundLogService.retreiveById(getSelectedDocOutboundLogIds().iterator().next());
+
+		final EDIExportStatus fromExportStatus = EDIExportStatus.ofCode(docOutboundLog.getEDI_ExportStatus());
+
+		return ChangeEDI_ExportStatusHelper.computeTargetExportStatusLookupValues(fromExportStatus);
+	}
+
+	@Override
+	@Nullable
+	public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
+	{
+		final I_C_Doc_Outbound_Log docOutboundLog = ediDocOutBoundLogService.retreiveById(getSelectedDocOutboundLogIds().iterator().next());
+
+		final EDIExportStatus fromExportStatus = EDIExportStatus.ofCode(docOutboundLog.getEDI_ExportStatus());
+
+		return ChangeEDI_ExportStatusHelper.computeParameterDefaultValue(fromExportStatus);
+	}
+
+	@Override
+	protected String doIt() throws Exception
+	{
+		final EDIExportStatus targetExportStatus = EDIExportStatus.ofCode(p_TargetExportStatus);
+
+		for (final ArchiveId logId : getSelectedDocOutboundLogIds())
```

Comment:
ArchiveId shall not be used for C_Doc_OutBound_Log_ID.
(note that we have a dedicated `AD_Archive` table that is also *referenced* by `C_Doc_Outbound_Log_Line`)

Reply:
> ArchiveId shall not be used for C_Doc_OutBound_Log_ID.

So should i add the ID class `DocOutboundLogId`?

Reply:
pls add `DocOutboundLogId`

Reply:
done